packages/core/spec/node/asciidoctor.spec.js

```diff
@@ -970,6 +970,18 @@ image::https://asciidoctor.org/images/octocat.jpg[GitHub mascot]`
       }
     })
 
+    it('should instantiate the specified logger class when the logger value is falsy', () => {
+      const defaultLogger = asciidoctor.LoggerManager.getLogger()
+      try {
+        const initialLogger = asciidoctor.LoggerManager.setLogger(null)
```

Comment:
I think you should treat the setLogger as void and follow with a separate call to `getLogger()`. In Ruby, it also takes two calls.

Reply:
Technically, `LoggerManager#logger=` Ruby method returns the value right?

https://github.com/asciidoctor/asciidoctor/blob/f580230151bdb8392dcc55d6e1d46bb57d858cf7/lib/asciidoctor/logging.rb#L93-L95

But I agree, it's a bit weird to get a value from a setter function.

Reply:
Setter methods always return nil in Ruby...even if you specify a return value.

Reply:
Interesting, it seems that Opal does things differently since `this['$logger=']` returns a value 🤔

Reply:
Actually, I just remembered. It's even weirder than that. Ruby always returns the value passed to the setter. In other words, it passes it through to act like any other assignment. Thus, it disregards any activity in the method itself.